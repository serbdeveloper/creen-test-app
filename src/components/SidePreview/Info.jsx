import React from 'react';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

const  Info = ({data}) => {
  const labelStyle = {
    flexGrow: 1,
    alignSelf: 'flex-end',
    textAlign: 'left',
    justifySelf: 'center',
    justifyContent: 'center',
    paddingLeft: 5,
    color: 'gray',
    width: '50px',
  }
  const valueStyle = {
    flexGrow: 1,
    alignSelf: 'flex-start',
    textAlign: 'left',
    justifySelf: 'center',
    justifyContent: 'center',
    paddingRight: 5,
    color: 'gray',
    width: '150px',
  }
  const gridItemStyle = {
    justifyContent: 'space-between',
    display: 'flex',
    paddingTop: 1,
    paddingBottom: 1,
    
  }
  const labelFontWeight = 500;
  const valueFontWeight = 400;
  
  return (
    <Grid  data-testid="info" fluid rowSpacing={3} columnSpacing={{ xs: 5, sm: 5, md: 5 }} sx={{ paddingTop: 1, paddingBottom: 1 }}>
      <Grid item xs={6} sx={gridItemStyle}>
          <Typography
            variant="subtitle2"
            noWrap
            component="div"
            fontWeight={400}
            sx={labelStyle}>
            Info
          </Typography>
        </Grid>

        <Grid item xs={6} sx={gridItemStyle}>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={labelFontWeight}
            sx={labelStyle}>
            Title
          </Typography>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={valueFontWeight}
            sx={valueStyle}>
            {data.title}
          </Typography>
        </Grid>

        <Grid item xs={6} sx={gridItemStyle}>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={labelFontWeight}
            sx={labelStyle}>
            Author
          </Typography>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={valueFontWeight}
            sx={valueStyle}>
            {data.nameOfAuthor}
          </Typography>
        </Grid>

        <Grid item xs={6} sx={gridItemStyle}>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={labelFontWeight}
            sx={labelStyle}>
            Year
          </Typography>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={valueFontWeight}
            sx={valueStyle}>
            {data.yearOfBublishing}
          </Typography>
        </Grid>

        <Grid item xs={6} sx={gridItemStyle}>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={labelFontWeight}
            sx={labelStyle}>
            Quantity
          </Typography>
          <Typography
            variant="h8"
            noWrap
            component="div"
            fontWeight={valueFontWeight}
            sx={valueStyle}>
            {data.quantity}
          </Typography>
      </Grid>
  </Grid>
  )
}

export default Info;