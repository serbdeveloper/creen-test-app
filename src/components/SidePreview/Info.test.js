import { render, screen } from "@testing-library/react";
import  Info  from "./Info";
import renderer from 'react-test-renderer';


const mockData = {
  "id": 1,
  "isbn": 1,
  "title": "Things Fall Apart",
  "nameOfAuthor": "Chinua Achebe",
  "dateOfBirthAuthor": "1930-11-16",
  "numberOfPages": 209,
  "yearOfPublishing": 1958,
  "quantity": 56,
  "coverPhoto": "https://book-store.mvsoft.co.rs/assets/images/things-fall-apart.jpg"
}


test('renders info component', () => {
  render(<Info data={mockData} />);
  const infoElement = screen.getByTestId('info');
  expect(infoElement).toBeInTheDocument();
});


describe('Info Snapshot', () => {
  it('Should matches DOM snapshot', () => {
    const tree = renderer.create(<Info data={mockData} />).toJSON()
    expect(tree).toMatchSnapshot();
  })
})
