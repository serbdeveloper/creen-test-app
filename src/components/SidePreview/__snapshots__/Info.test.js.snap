// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Snapshot Should matches DOM snapshot 1`] = `
<div
  className="MuiGrid-root css-kdmkmi-MuiGrid-root"
  data-testid="info"
  fluid={true}
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-13wm5bs-MuiGrid-root"
  >
    <div
      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-y50ajl-MuiTypography-root"
    >
      Info
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-13wm5bs-MuiGrid-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-142fl07-MuiTypography-root"
    >
      Title
    </div>
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-cj8e0d-MuiTypography-root"
    >
      Things Fall Apart
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-13wm5bs-MuiGrid-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-142fl07-MuiTypography-root"
    >
      Author
    </div>
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-cj8e0d-MuiTypography-root"
    >
      Chinua Achebe
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-13wm5bs-MuiGrid-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-142fl07-MuiTypography-root"
    >
      Year
    </div>
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-cj8e0d-MuiTypography-root"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-13wm5bs-MuiGrid-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-142fl07-MuiTypography-root"
    >
      Quantity
    </div>
    <div
      className="MuiTypography-root MuiTypography-h8 MuiTypography-noWrap css-cj8e0d-MuiTypography-root"
    >
      56
    </div>
  </div>
</div>
`;
